packages/maskbook/src/components/InjectedComponents/TypedMessageRenderer.tsx

```diff
@@ -92,14 +93,40 @@ registerTypedMessageRenderer('anchor', {
     priority: 0,
 })
 
+function isBase64(str: string) {
+    if (str === '' || str.trim() === '') { return false; }
+    try {
+        return btoa(atob(str)) == str;
+    } catch (err) {
+        return false;
+    }
+}
+
 export const DefaultTypedMessageImageRenderer = memo(function DefaultTypedMessageImageRenderer(
     props: TypedMessageRendererProps<TypedMessageImage>,
```

Comment:
As you can see, it receives a `TypedMessageRendererProps<TypedMessageImage>` which means `props.message` is a `TypedMessageImage` and you should use the info in `TypedMessageImage` to render the image.